tests/nested/manual/core-early-config/task.yaml

```diff
@@ -0,0 +1,79 @@
+summary: Test that config defaults are applied early when image is created.
+
+systems: [ubuntu-18.04-64]
+
+prepare: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    mkdir extra-snaps
+
+    snap download --channel=18/stable pc
+    GADGET_SNAP=$(ls pc_*.snap)
+    unsquashfs -no-progress "$GADGET_SNAP"
+    rm -f "$GADGET_SNAP"
+
+    cat defaults.yaml >> squashfs-root/meta/gadget.yaml
+
+    mkdir -p squashfs-root/meta/hooks
+    cp install squashfs-root/meta/hooks/
+    mksquashfs squashfs-root "$GADGET_SNAP" -comp xz -no-fragments -no-progress
+    rm -rf squashfs-root
+    mv "$GADGET_SNAP" extra-snaps/
+
+    # XXX: pc-kernel initrd fix
+    snap download --channel=18/stable pc-kernel
+    KERNEL_SNAP=$(ls pc-kernel_*.snap)
```

Comment:
Well, here could be a problem, as part of the nested vm creation we are doing this:
. "$TESTSLIB"/prepare.sh
snap download --basename=pc-kernel --channel="20/edge" pc-kernel
uc20_build_initramfs_kernel_snap "$PWD/pc-kernel.snap" "$WORK_DIR/image"
EXTRA_FUNDAMENTAL="--snap $WORK_DIR/image/pc-kernel_*.snap"

So we are adding 2 pc-kernel as extra snap and not sure which is used. For sure you should call uc20_build_initramfs_kernel_snap and see it that helps.

Reply:
Sorry, this is just for uc20